hudi-common/src/main/java/org/apache/hudi/common/util/HoodieAvroUtils.java

```diff
@@ -214,6 +220,63 @@ private static GenericRecord rewrite(GenericRecord record, Schema schemaWithFiel
     return newRecord;
   }
 
+  /*
+  This function takes the union of all the fields except hoodie metadata fields
+   */
+  private static List<Field> getAllFieldsToWrite(Schema oldSchema, Schema newSchema) {
+    Set<Field> allFields = new HashSet<>(oldSchema.getFields());
+    List<Field> fields = new ArrayList<>(oldSchema.getFields());
+    for (Schema.Field f : newSchema.getFields()) {
+      if (!allFields.contains(f) && !isMetadataField(f.name())) {
+        fields.add(f);
+      }
+    }
+
+    return fields;
+  }
+
+  private static void populateNewRecordAsPerDataType(GenericRecord record, Field field) {
```

Comment:
nit: May be rename this to `populateFieldWithDefaultValue` as that seems to be the intent of this function.

Reply:
Removed this function also.